### DISCUSSION TITLE
How do I Login With OIDC

### DISCUSSION THREAD

**Discussion by ShiromMakkad** [LeoKlaus/plappa#230] [Q&A] [answered]
When I select Audiobookshelf as an option, I'm asked for my username and password, but I don't use password auth to login. Is there a way to login with OIDC?

**Comment by LeoKlaus** [comment]
Hi Shirom,
Yes, you have enter the address of you server first. Plappa then queries the supported login types and displays buttons for those (including the OIDC button).

**Reply by ShiromMakkad** [reply]
I put in my address but nothing happens

**Reply by LeoKlaus** [reply]
Can you make sure that https://your.abs.server/status does return a JSON response _without_ any additional authorization?
You can just use this curl command to test that:
``` bash
curl --request GET \
  --url https://abs.your.domain/status
```

The response should look like this:
``` json
{
	"app": "audiobookshelf",
	"serverVersion": "2.16.0",
	"isInit": true,
	"language": "en-us",
	"authMethods": [
		"local",
		"openid"
	],
	"authFormData": {
		"authLoginCustomMessage": null,
		"authOpenIDButtonText": "Login with OpenId",
		"authOpenIDAutoLaunch": false
	}
}
```

**Reply by mrburns-42** [reply]
In Audiobookshelf you must also specify that Plappa is allowed for SSO. You do this by adding plappa://oauth

For me it looks like this in the settings in ABS (in German):

<img width="882" alt="Bildschirmfoto 2024-12-12 um 23 31 02" src="https://github.com/user-attachments/assets/ccf81063-3dc1-4f20-ad1f-06aee61bf77c" />

Maybe that will help.

**Reply by ShiromMakkad** [reply, marked as answer]
Sorry it's been a while. My server's wifi went out while I was out of town. Anyways, the original issue was that I forgot to turn off my nginx auth when I turned on OIDC. So I wasn't able to access that url originally like you predicted. Now I see the login with oidc button. Ngl, I'd prefer that the button was always visible always. Either way, now that it's documented it should be googleable. 

And yea you do need `plappa://oauth` in your audiobookshelf settings. Also, you need `*` in the allowed redirect URIs for your OIDC provider. If you use `plappa://oauth` there, you get "Redirect URI Error," at least for authentik. I don't think plappa sends the redirect URI to the OIDC provider. This would be a nice feature to comply with some OIDC providers' security features.

**Reply by manicmoose** [reply]
I found that regex: '.*' is required by Authentik.
So yeah, would be nice if Plappa could fix the redirect URI properly.

**Reply by tebriel** [reply]
Adding context for someone coming along later:
My only authentik redirect URIs are the following (both strict):
- `https://<base-url>/auth/openid/callback`
- `https://<base-url>/auth/openid/mobile-redirect`

`.*` would be highly permissive I think this should work instead.
